<a href="https://colab.research.google.com/github/galihtw04/data-science-2026/blob/master/Pertemuan3_Muhammad_Galih_Abdurrahman_240401010278.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Pertemuan 3
Nama : Muhammad Galih Abdurrahman

NIM : 240401010278

Kelas IF403

In [1]:
import pandas as pd
import numpy as np
from scipy.stats.mstats import winsorize

# Muat Dataset
url = 'https://drive.usercontent.google.com/download?id=1LfQWProB0VjWN5q8bKuRIgn-stULfIRo&export=download&authuser=0'
df = pd.read_csv(url)

In [2]:
# Eksplorasi Info, Describe, dan Missing
print('\nInfo dataset: \n',df.info())
print('\nStatistik dataset: \n',df.describe())
print('\nTotal missing: \n',df.isnull().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130 entries, 0 to 129
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            130 non-null    int64  
 1   luas_m2       112 non-null    float64
 2   harga_juta    113 non-null    float64
 3   kota          130 non-null    object 
 4   kamar         120 non-null    float64
 5   tahun_bangun  130 non-null    int64  
 6   kondisi       130 non-null    object 
dtypes: float64(3), int64(2), object(2)
memory usage: 7.2+ KB

Info dataset: 
 None

Statistik dataset: 
                id      luas_m2    harga_juta       kamar  tahun_bangun
count  130.000000   112.000000  1.130000e+02  120.000000    130.000000
mean    65.500000   267.627679  8.856325e+05    3.433333   2062.638462
std     37.671829   885.664181  9.407144e+06    1.776283    701.684043
min      1.000000   -50.000000 -5.000000e+02    1.000000   1890.000000
25%     33.250000    87.050000  3.450000e+02    

In [9]:
# Hapus Baris Duplikat
df.drop_duplicates(inplace=True)
print('Jumlah duplikat:', df.duplicated().sum())

Jumlah duplikat: 0


In [10]:
# Normalisasi
df['kota'] = df['kota'].str.strip().str.title()
df['kondisi'] = df['kondisi'].str.strip().str.lower()

In [11]:
# Imputasi Missing Values
df['luas_m2'] = df['luas_m2'].fillna(df['luas_m2'].median())
df['harga_juta'] = df['harga_juta'].fillna(df['harga_juta'].median())
df['kamar'] = df['kamar'].fillna(df['kamar'].mode() [0])

In [12]:
# Penanganan Outliers
for col in ['harga_juta', 'luas_m2', 'tahun_bangun']:
  Q1, Q3 = df[col].quantile([0.25, 0.75])
  IQR = Q3 -Q1
  df[col] = df[col].clip(Q1-1.5*IQR, Q3+1.5*IQR)

In [13]:
assert df.isnull().sum().sum() == 0, 'Masih ada missing!'
assert df.duplicated().sum().sum() == 0, 'Masih ada duplikat!'
print("Total missing values:", df.isnull().sum().sum())
print("Total duplikat:", df.duplicated().sum())
print('Shape akhir:', df.shape)

Total missing values: 0
Total duplikat: 0
Shape akhir: (130, 7)


In [14]:
df.to_csv('housing_clean.csv', index=False)
print('Dataset bersih telah tersimpan!')

Dataset bersih telah tersimpan!


## Kesimpulan

Pada Pertemuan 3 ini, saya mempelajari proses data cleaning atau pembersihan data menggunakan Python, khususnya dengan library Pandas dan NumPy. Dataset yang digunakan adalah dataset housing yang terdiri dari 130 baris dan 7 kolom. Dari hasil eksplorasi awal, ditemukan beberapa permasalahan pada data, seperti missing value pada kolom `luas_m2`, `harga_juta`, dan `kamar`, serta adanya nilai yang tidak wajar atau outlier pada beberapa kolom numerik.

Langkah pembersihan data dilakukan dengan beberapa tahapan, yaitu menghapus data duplikat, menormalisasi format teks pada kolom `kota` dan `kondisi`, mengisi nilai kosong menggunakan median dan modus, serta menangani outlier menggunakan metode IQR dengan teknik clipping. Setelah proses cleaning selesai, dilakukan validasi ulang untuk memastikan tidak ada missing value dan duplikasi data yang tersisa. Hasil akhirnya menunjukkan bahwa total missing value menjadi 0, total duplikat menjadi 0, dan bentuk akhir dataset tetap 130 baris dan 7 kolom.

Temuan utama dari pertemuan ini adalah bahwa data mentah sering kali memiliki masalah seperti nilai kosong, format teks tidak konsisten, duplikasi, dan outlier. Masalah tersebut perlu ditangani sebelum data digunakan untuk analisis atau machine learning agar hasil yang diperoleh lebih akurat dan dapat dipercaya.

Keterbatasan dari proses ini adalah penanganan outlier masih dilakukan secara umum menggunakan batas IQR, sehingga belum mempertimbangkan konteks bisnis atau kondisi nyata dari data properti. Pertanyaan lanjutan yang muncul adalah bagaimana hasil analisis atau model prediksi akan berubah jika metode penanganan outlier dan imputasi yang digunakan berbeda.